src/internal_events/kubernetes/instrumenting_watcher.rs

```diff
@@ -26,6 +26,21 @@ impl<E: Debug> InternalEvent for WatchRequestInvocationFailed<E> {
     }
 }
 
+#[derive(Debug)]
+pub struct WatchStreamFailed<E> {
+    pub error: E,
+}
+
+impl<E: Debug> InternalEvent for WatchStreamFailed<E> {
+    fn emit_logs(&self) {
+        error!(message = "Watch stream failed.", error = ?self.error, internal_log_rate_secs = 5);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("k8s_watch_stream_failed_total", 1);
```

Comment:
I think we'll want to add this to the docs too: https://github.com/timberio/vector/blob/1e54aca6970a3f141749bd56d8cd8c98ba6e3fab/docs/reference/components/sources/kubernetes_logs.cue#L567-L574

Reply:
On a quick glance I think the docs are missing several metrics, do you want me to open another issue or just add them while I'm here?

Reply:
😢 if you don't mind adding them while you are looking, that would be nice. It could be a separate PR though, if you prefer to just add this one for now.

Eventually I think it'd be cool to extract these metrics from the rust code.